pkg/kubelet/certificate/kubelet.go

```diff
@@ -221,6 +245,27 @@ func NewKubeletClientCertificateManager(
 	)
 	legacyregistry.Register(certificateRenewFailure)
 
+	usages := []certificates.KeyUsage{
+		// https://tools.ietf.org/html/rfc5280#section-4.2.1.3
+		//
+		// DigitalSignature allows the certificate to be used to verify
+		// digital signatures including signatures used during TLS
+		// negotiation.
+		certificates.UsageDigitalSignature,
+		// ClientAuth allows the cert to be used by a TLS client to
+		// authenticate itself to the TLS server.
+		certificates.UsageClientAuth,
+	}
+	if cert, err := certificateStore.Current(); err == nil {
```

Comment:
same comment here, make the usages dynamic based on the key rather than calling certificateStore.Current()